tests/unit/test_session.py

```diff
@@ -588,7 +588,7 @@ def test_transform_pack_to_request(sagemaker_session):
 
     sagemaker_session.transform(job_name=JOB_NAME, model_name=model_name, strategy=None, max_concurrent_transforms=None,
                                 max_payload=None, env=None, input_config=in_config, output_config=out_config,
-                                resource_config=resource_config, tags=None)
+                                resource_config=resource_config, tags=None, data_processing=None)
```

Comment:
You need to unit test the `transform` function with different `data_processing` use cases.

Reply:
Nice catch. done. There are two tests 1 with required parameter and other with optional parameter to cover different use case. I used them accordingly https://github.com/aws/sagemaker-python-sdk/pull/864